src/main/java/cn/nukkit/Player.java

```diff
@@ -4554,6 +4554,11 @@ private void onBlockBreakComplete(BlockVector3 blockPos, BlockFace face) {
             handItem = this.level.useBreakOn(blockPos.asVector3(), face, handItem, this, true);
             if (handItem == null) {
                 this.level.sendBlocks(new Player[]{this}, new Vector3[]{blockPos.asVector3()}, UpdateBlockPacket.FLAG_ALL_PRIORITY);
+
+                BlockEntity blockEntity = this.level.getBlockEntity(blockPos.asVector3());
+                if (blockEntity instanceof BlockEntitySpawnable) {

```

Comment:
I think it can be done directly like this
```
if (blockEntity instanceof BlockEntitySpawnable blockEntitySpawnable) {
    blockEntitySpawnable.spawnTo(this);
}
```

Reply:
I don't see the need to create an extra variable if it will only be used once.